### PR TITLE
Api endpoints

### PR DESCRIPTION
**Pull request by devynlab**
- Check for duplicates when saving a new API with endpoints and only save the first on, returning the duplicates in an array
- Documenting **CreateEndpointDto**

### CODE REVIEW

File: src/api/api.service.ts [hunk 1]
```diff
@@ -231,23 +231,30 @@ export class ApiService {
...
     const duplicateEndpoints = [];
-    for (const endpoint of newEndpoints) {
+    const endpointsToSave = [];
+    for (const createEndpointDto of createEndpointsDto) {
+      const endpoint = this.endpointsRepo.create({
+        ...createEndpointDto,
+        apiId,
+      });
       const endpointKey = `${endpoint.method} ${endpoint.route}`;
       if (endpointSet.has(endpointKey)) {
```
Comment (PhilemonBrain) [inline src/api/api.service.ts:243]: @devynlab i think this is better    
      if (!endpointSet.has(endpointKey)) {
          endpointSet.add(endpointKey);
          endpointsToSave.push(endpoint);
        }

File: src/common/interfaces/endpoint-impl.interface.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+  @IsOptional()
+  @ApiProperty({ type: ReqBodyClass })
+  value: ReqBody['value'];
+
+  @IsBoolean()
+  @ApiProperty()
+  required: boolean;
+
+  constructor(
```
Comment (PhilemonBrain) [inline src/common/interfaces/endpoint-impl.interface.ts:58]: wow, i havent seen this b4 @devynlab. are you sure its not prone to bugs?

File: src/common/interfaces/endpoint-impl.interface.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  })
+  value: string | Date | boolean | number | object | symbol | Array<any>;
+
+  constructor(key: string, value: ReqBody['value']) {
+    this.key = key;
+    this.value = value;
+  }
+}
+
+export class HeaderTypeClass implements HeaderType {
```
Comment (PhilemonBrain) [inline src/common/interfaces/endpoint-impl.interface.ts:36]: @devynlab why are you implementing an interface

**Review by PhilemonBrain — CHANGES_REQUESTED** [review]

File: src/common/interfaces/endpoint-impl.interface.ts (hunk 2 above)
Comment (devynlab) [reply src/common/interfaces/endpoint-impl.interface.ts:58]: I have tested and it works just fine, plus the added validation. 
But of course I'll need you guys to also just validate with your tests...

File: src/common/interfaces/endpoint-impl.interface.ts (hunk 3 above)
Comment (devynlab) [reply src/common/interfaces/endpoint-impl.interface.ts:36]: I had to coz of how Swagger interacts with interfaces.., there is no way to document interfaces directly, it leads to some issues at runtime...

File: src/api/api.service.ts (hunk 1 above)
Comment (devynlab) [reply src/api/api.service.ts]: Done

**Review by PhilemonBrain — APPROVED** [review]

**Pull request merged**